crates/walrus-service/src/node.rs

```diff
@@ -2204,14 +2222,24 @@ impl StorageNodeInner {
         Ok(metadata)
     }
 
+    /// Returns the latest event epoch.
+    /// The callsite must ensure that the first event is processed so that the
+    /// latest_event_epoch_watcher is initialized.
     fn current_event_epoch(&self) -> Epoch {
-        self.latest_event_epoch.load(Ordering::SeqCst)
+        self.latest_event_epoch_sender
```

Comment:
Question: Don't we need the `latest_event_epoch_watcher` here?

Reply:
Good catch! I think they refer to the same underlying value, but I should use `latest_event_epoch_watcher` for reads.